[PATCH v17] add sample shippable.yml and Dockerfile for Aarch64 builds

ttrahan opened the pull request:
Adds shippable.yml config file for testing ODP builds on Aarch64 hardware using Shippable. This CI config also uses a Dockerfile to install additional dependencies on top of the standard Shippable Aarch64 image.

muvarov commented on Dockerfile:
```diff
@@ -0,0 +1,23 @@
+    fi
+
+RUN sudo apt-get update && sudo apt-get install -yy \
+  gcc-4.8 \
```
this needs to be alphabetic sorted.

muvarov commented on shippable.yml:
```diff
@@ -0,0 +1,81 @@
+    # - CROSS_ARCH="armhf" CFLAGS="-march=armv7-a"
+    # - CROSS_ARCH="powerpc"
+    # - CROSS_ARCH="i386"
+  global:
```
Coverity I also thing should go away. This script does not send results to Coverity.

ttrahan commented:
@muvarov : your requested changes have been applied.

muvarov commented:
@ttrahan please squash all patches. We use a little bit non standard procedure. From github we generate patches and send them to mailing list for review and discussion. Each patch should standalone self contained fix.

ttrahan commented:
@muvarov: squashed into single commit. 

Note, one of the four matrix builds in this sample is failing due to a failed test:
https://bugs.linaro.org/show_bug.cgi?id=3343

ttrahan commented:
The failed test issue is now resolved and no longer failing. Comment added to https://bugs.linaro.org/show_bug.cgi?id=3343.

lumag commented on shippable.yml:
```diff
@@ -0,0 +1,62 @@
+
+  on_failure:
+    - cat config.log
+    - find . -name 'test-suite.log' -execdir grep -il "FAILED" {} \; -exec echo {} \; -exec cat {} \;
```
Please use command from updated .travis.yml

lumag commented on shippable.yml:
```diff
@@ -0,0 +1,62 @@
+    - sudo mkdir -p /mnt/huge
+    - sudo mount -t hugetlbfs nodev /mnt/huge
+    - mkdir -p /dev/shm/odp
+    - export CUNIT_VERSION=2.1-3
```
This should not be necessary. Ubuntu 16.04 already provides CUnit 2.1-3. So it is better to use system one.

ttrahan commented:
Per @lumag comments, installation and config of CUNIT has been removed, and newest "on_failure" command from .travis.yml added.

lumag commented on shippable.yml:
```diff
@@ -0,0 +1,56 @@
+      autoconf
+      ./configure --prefix=$HOME/cunit-install --enable-debug --enable-automated --enable-basic --enable-console --enable-examples --enable-test || cat config.log
+      make
+      sudo make install
```
This whole block (lines 38--46) is a part of CUnit handling and so should be dropped

lumag commented on scripts/Dockerfile:
```diff
@@ -0,0 +1,23 @@
+  graphviz \
+  kmod \
+  mscgen \
+  libpcap-dev \
```
```diff
+ libcunit1-dev \
```

lumag commented on shippable.yml:
```diff
@@ -0,0 +1,56 @@
+      sudo make install
+    - ./bootstrap
+    - export PKG_CONFIG_PATH="$HOME/cunit-install/lib/pkgconfig:${PKG_CONFIG_PATH}"
+    - export LD_LIBRARY_PATH="$HOME/cunit-install/lib:$LD_LIBRARY_PATH"
```
Also you can drop these two lines. Again, CUnit handling

lumag commented on shippable.yml:
```diff
@@ -0,0 +1,56 @@
+    - export LD_LIBRARY_PATH="$HOME/cunit-install/lib:$LD_LIBRARY_PATH"
+    - ./configure --disable-test-perf
+    - make
+    - sudo LD_LIBRARY_PATH="$HOME/cunit-install/lib:$LD_LIBRARY_PATH" ODP_SHM_DIR=/dev/shm/odp make check
```
Replace with `sudo env ODP_SHM_DIR=/dev/shm/odp make check`

ttrahan commented:
Per @lumag comments: all changes applied
* Removed all references to manual CUNIT config
* Added libcunit1-dev package to Dockerfile for install
* Updated make check statement

CI running here to validate: 
https://app.shippable.com/github/ttrahan/odp/runs/10/summary/console

lumag commented on shippable.yml:
```diff
@@ -0,0 +1,45 @@
+    # - CONF="--enable-schedule-sp"
+    # - CONF="--enable-schedule-iquery"
+    # - CONF="--enable-dpdk-zero-copy"
+    - CROSS_ARCH="arm64"
```
It seems there is a confusion here. Currently this setup generates 4 build jobs (2 for CONF setting, 2 for CROSS_ARCH). There is no need to run CROSS_ARCH jobs (and it is not a cross-build anyway). Let's drop CROSS_ARCH settings altogether.

ttrahan replied:
My understanding is that this shippable.yml will evolve shortly to perform the same build functionality in the .travis.yml config, including the cross-arch testing. This is simply the initial build sample to start with a simple example, that will then proceed to add in the additional tests. As such, I suggest leaving this in as a placeholder for these next steps.

muvarov replied:
yep, I agree to have it as it now. We need to make some simple run. Then extend it. It will be nice if we can have common file for Shippable and Travis or generate them from some common template. But that can be later.

lumag replied:
@ttrahan @muvarov ATM this uncomented CROSS_ARCH lines does not make sense. I'd prefer to see it commented, so that there are no extra builds in Shippable.

ttrahan replied:
@lumag: agreed. I've commented out the unused CROSS_ARCH variable and the unused global variable.

lumag replied:
@ttrahan Thanks! It now looks fine for me.

muvarov commented:
@ttrahan sorry for delay, we repaired web form to add new odp contributor sign license, but did not update web site. Web man is off today so I think tomorrow everything has to work and we can continue.

muvarov commented:
@ttrahan  please add your self to odp contributors https://www.opendataplane.org/contributor/individual/ looks like link works now.

ttrahan commented:
@muvarov: I've added myself to odp contributors

muvarov commented:
@ttrahan Tom,  license is now ok. Commit has to have sign-off.  (use git commit --amend -s to add it automatically). I trailing while spaces. It's better to remove them. Other than that patch is ok.

ttrahan commented:
@muvarov: Commit is signed off and trailing whitespaces removed. Thanks for you patience on this as I get up to speed.

muvarov commented:
@ttrahan  do not see sign-off here: https://github.com/Linaro/odp/pull/231.patch

ttrahan commented:
@muvarov: Sorry about that. Now resolved.

muvarov commented:
Merged.
